pkg/admissioncontroller/webhooks/add.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhooks
+
+import (
+	"context"
+
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"github.com/gardener/gardener/pkg/admissioncontroller/apis/config"
+	"github.com/gardener/gardener/pkg/admissioncontroller/webhooks/admission/auditpolicy"
+	"github.com/gardener/gardener/pkg/admissioncontroller/webhooks/admission/internaldomainsecret"
+	"github.com/gardener/gardener/pkg/admissioncontroller/webhooks/admission/kubeconfigsecret"
+	"github.com/gardener/gardener/pkg/admissioncontroller/webhooks/admission/namespacedeletion"
+	"github.com/gardener/gardener/pkg/admissioncontroller/webhooks/admission/resourcesize"
+	"github.com/gardener/gardener/pkg/admissioncontroller/webhooks/admission/seedrestriction"
+	seedauthorizer "github.com/gardener/gardener/pkg/admissioncontroller/webhooks/auth/seed"
+	seedauthorizergraph "github.com/gardener/gardener/pkg/admissioncontroller/webhooks/auth/seed/graph"
+)
+
+// AddWebhookHandlersToManager adds all webhook handlers to the given manager.
+func AddWebhookHandlersToManager(
+	ctx context.Context,
+	mgr manager.Manager,
+	cfg *config.AdmissionControllerConfiguration,
+) error {
+	var (
+		log         = mgr.GetLogger().WithName("webhook")
+		logSeedAuth = log.WithName(seedauthorizer.AuthorizerName)
+		server      = mgr.GetWebhookServer()
+	)
+
+	graph := seedauthorizergraph.New(mgr.GetLogger().WithName("seed-authorizer-graph"), mgr.GetClient())
```

Comment:
Shouldn't this be:
```suggestion
	graph := seedauthorizergraph.New(log.WithName("seed-authorizer-graph"), mgr.GetClient())
```

Reply:
Initially, I had done it like this, however I found it confusing that the graph logs would be under a "webhook" logger name. The graph itself is no webhook, it's just an "internal tool" used by one of the webhook handler, so I chose to only name it "seed-authorizer-graph".